examples/graphics/mesh-decals.html

```diff
@@ -0,0 +1,245 @@
+
+        // Allocate storage for colors, each vertex stores r, g, b and a
+        this.colors = new Uint8ClampedArray(4 * numDecalVertices);
+
+        // Allocate storage for uvs, each vertex stores u and v. And fill them up to display whole texture
+        this.uvs = [];
+        var i;
+        for (i = 0; i < this.numDecals; i++)
+            uvs.push(0, 0,      0, 1,       1, 1,       1, 0);
+
+        // Allocate and generate indices. Each quad is representing using 2 triangles, and uses 4 vertices
+        var quadTriangles = [0, 1, 2,       2, 3, 0];
+        this.indices = new Uint16Array(6 * this.numDecals);
+        for (i = 0; i < this.numDecals; i++) {
+            this.indices[6 * i + 0] = 4 * i + quadTriangles[0];     
+            this.indices[6 * i + 1] = 4 * i + quadTriangles[1];
+            this.indices[6 * i + 2] = 4 * i + quadTriangles[2];
+            this.indices[6 * i + 3] = 4 * i + quadTriangles[3];
+            this.indices[6 * i + 4] = 4 * i + quadTriangles[4];
+            this.indices[6 * i + 5] = 4 * i + quadTriangles[5];
+        }
+
+        // Helper function to generate a decal with index i at position pos. It fills up information for all 4 vertices of a quad
+        function CreateDecal(i, pos) {
+
+            // random size and rotation angle
+            var size = 0.5 + Math.random();
+            var angle = Math.random() * Math.PI;
+
+            // random color
+            var r = Math.random() * 255;
+            var g = Math.random() * 255;
+            var b = Math.random() * 255;
+
+            var j;
+            for (j = 0; j < 4; j++) {
+                this.colors[i * 16 + j * 4 + 0] = r;
+                this.colors[i * 16 + j * 4 + 1] = g;
+                this.colors[i * 16 + j * 4 + 2] = b;
+                this.colors[i * 16 + j * 4 + 3] = 0;    // alpha is not used by shader
+            }
+
+            // vertex positions to form a square quad with random rotation and size
+            this.positions[12 * i + 0] = pos.x + size * Math.sin(angle);      this.positions[12 * i + 1] = 0;     this.positions[12 * i + 2] = pos.z + size * Math.cos(angle);      angle += Math.PI * 0.5;
+            this.positions[12 * i + 3] = pos.x + size * Math.sin(angle);      this.positions[12 * i + 4] = 0;     this.positions[12 * i + 5] = pos.z + size * Math.cos(angle);      angle += Math.PI * 0.5;
+            this.positions[12 * i + 6] = pos.x + size * Math.sin(angle);      this.positions[12 * i + 7] = 0;     this.positions[12 * i + 8] = pos.z + size * Math.cos(angle);      angle += Math.PI * 0.5;
+            this.positions[12 * i + 9] = pos.x + size * Math.sin(angle);      this.positions[12 * i + 10] = 0;    this.positions[12 * i + 11] = pos.z + size * Math.cos(angle);     angle += Math.PI * 0.5;
+        }
+
+        // helper function to update required vertex streams
+        function UpdateMesh(device, mesh, updatePositions, updateColors, initAll) {
```

Comment:
UpdateMesh -> updateMesh (stick to camel case)

Reply:
fixing, thanks